ibazel/fs_watcher_darwin.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build darwin
+
+package main
+
+import (
+	"os"
+
+	"github.com/bazelbuild/bazel-watcher/ibazel/fs_watcher"
+	"github.com/bazelbuild/bazel-watcher/ibazel/fsnotify_watcher"
+	"github.com/bazelbuild/bazel-watcher/ibazel/fsevents_watcher"
+)
+
+func NewWatcher() (fs_watcher.Watcher, error) {
+	flag, ok := os.LookupEnv("IBAZEL_USE_LEGACY_WATCHER") 
```

Comment:
@achew22 I'm open to alternatives here if think it it would be better to default to the legacy, but as far as I can tell, its a positive to always use FSEvents over kqueue on macOS.

Reply:
What would you think about making it opt in for a release or two and then letting it soak? I'm pretty sure that no one really beta tests this software or reads my release notes, but in case someone does it would be nice to try it out on a couple of projects before things are completely replaced.

Reply:
Actually, I had originally written it where it was opt-in, but then was thinking that likely no one would ever try to use it or know it was there; and therefore we might not get *any* feedback on it. If integrated properly FSEvents should only be a positive on macOS which is why I made it a default. I figured if people encountered issues, they could create tickets for issues, and set the var to use the legacy watcher until its fixed. I'll do whatever you think is best, but was worried that we wouldn't get any results from 'letting it soak' and it would just delay turning it on for all macOS.